### PR TITLE
Submarines

### PR DESCRIPTION
**Pull request by EdenZadikove**
Issues: #5 and #6

### CODE REVIEW

File: Project/Code/FinalProject/Client/src/main/java/org/hit/internetprogramming/eoh/client/action/MenuAction.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
     GET_REACHABLES("Get Reachable Vertices", ActionType.GET_REACHABLES),
     CONNECTED_COMPONENTS("Get Connected Components ", ActionType.CONNECTED_COMPONENTS),
     SHORTEST_PATHS("Find Shortest Paths ", ActionType.SHORTEST_PATHS),
+    SUBMARINES("Find submarines in a graph", ActionType.SUBMARINES),
```
Comment (haimadrian) [inline Project/Code/FinalProject/Client/src/main/java/org/hit/internetprogramming/eoh/client/action/MenuAction.java]: Please modify it to "Count submarines" cause we don't return the submarines, only the amount of submarines

File: Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/action/ActionType.java [hunk 2]
```diff
@@ -69,7 +69,10 @@
      * Find all shortest paths, in a weighted graph, between a source vertex and destination vertex
      */
     SHORTEST_PATHS_IN_WEIGHTED_GRAPH,
-
+    /**
```
Comment (haimadrian) [inline Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/action/ActionType.java:73]: Please add a line break before this line

File: Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/action/ActionType.java [hunk 3]
```diff
@@ -69,7 +69,10 @@
      * Find all shortest paths, in a weighted graph, between a source vertex and destination vertex
      */
     SHORTEST_PATHS_IN_WEIGHTED_GRAPH,
-
+    /**
+     * Calculates how many submarines there are in a graph.
+     * */
+    SUBMARINES,
```
Comment (haimadrian) [inline Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/action/ActionType.java:76]: And another line break after this line. (Keep the format of the file)

File: Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/graph/IGraph.java [hunk 4]
```diff
@@ -88,8 +88,8 @@
     */
    List<T> getVertices();
 
-   /**
-    * Get all edges in this graph.<br/>
+   /**t all edges
+    * Gein this graph.<br/>
```
Comment (haimadrian) [inline Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/graph/IGraph.java]: ????
Revert this change

File: Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/mat/Submarine.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ * @see IMatrix
+ */
+public class Submarine {
+    private int numOfIndices;
+    private List<Index> connectedComponents= new ArrayList<>();
+
+    public Submarine() {}
+
+
+}
```
Comment (haimadrian) [inline Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/mat/Submarine.java]: Where do you use this class? Don't you need to delete it?

**Review by haimadrian — CHANGES_REQUESTED** [review]

File: Project/Code/FinalProject/Client/src/main/java/org/hit/internetprogramming/eoh/client/action/MenuAction.java (hunk 1 above)
Comment (EdenZadikove) [reply Project/Code/FinalProject/Client/src/main/java/org/hit/internetprogramming/eoh/client/action/MenuAction.java]: done

File: Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/action/ActionType.java (hunk 2 above)
Comment (EdenZadikove) [reply Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/action/ActionType.java:73]: done

File: Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/action/ActionType.java (hunk 3 above)
Comment (EdenZadikove) [reply Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/action/ActionType.java:76]: done

File: Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/graph/IGraph.java (hunk 4 above)
Comment (EdenZadikove) [reply Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/graph/IGraph.java]: done

File: Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/mat/Submarine.java (hunk 5 above)
Comment (EdenZadikove) [reply Project/Code/FinalProject/Common/src/main/java/org/hit/internetprogramming/eoh/common/mat/Submarine.java]: I do, done

**Review by haimadrian — APPROVED** [review]
